components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/internal/APIResourceManagementServiceComponent.java

```diff
@@ -127,4 +136,25 @@ protected void unsetIdentityEventService(IdentityEventService identityEventServi
         LOG.debug("IdentityEventService unset in API Resource Management bundle.");
     }
 
+    /**
+     * Checks if RAR is enabled by verifying the existence of required database tables.
+     * <p>
+     * This method iterates over a predefined list of database table names necessary for RAR functionality.
+     * If all tables exist, the method returns {@code true}.
+     * </p>
+     *
+     * @return {@code true} if all required tables for RAR are present in the database; {@code false} otherwise.
+     */
+    private boolean isRichAuthorizationRequestsEnabled() {
+
+        for (final String tableName : AuthorizationDetailsTypesUtil.RICH_AUTHORIZATION_REQUESTS_TABLES) {
+            if (!IdentityDatabaseUtil.isTableExists(tableName)) {
```

Comment:
don't we have a configuration to enable this feature? Why are we relying on the table exists?

Reply:
This has added to prevent database exceptions such as "Table does not exist"

Reply:
We decided to not use isTableExist checks in the product due to schema owner issues in oracle DB. Shall we use a configuration and properly document the need of necessary tables before enabling the configuration.

Reply:
@madurangasiriwardena instead of using 
```
isTableExists
```
method can we have a configuration.
@VimukthiRajapaksha correct me if I am wrong, this config is needed to seperate the feature for any specific deployments right?

Reply:
Since this is a feature going in a product release, all the required new database tables are supposed to be added with the product upgrade procedure. So there is no requirement to avoid "Table does not exist" exceptions.

If there are no other reasons to enable/disable the feature, we can get rid of this altogether.

Reply:
@VimukthiRajapakshaLet's address this, and then we can proceed.